utils/python/edit_versions_yaml.py

```diff
@@ -0,0 +1,53 @@
+import yaml
+import optparse
+import os
+
+VERSIONS_PATH = os.path.dirname(os.path.realpath(__file__)) + "/../../provision/acc_provision/versions.yaml"
+
+def read_yaml_file(file):
+    with open(file, 'r') as stream:
+        try:
+            orig_data = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+    return orig_data
+
+def modify_yaml(opt0, opt1, opt2, opt3, data):
+    data['versions'][float(opt0)]['openvswitch_version'] = opt3
+    data['versions'][float(opt0)]['opflex_agent_version'] = opt2
+    data['versions'][float(opt0)]['cnideploy_version'] = opt1
+    data['versions'][float(opt0)]['aci_containers_controller_version'] = opt1
+    data['versions'][float(opt0)]['aci_containers_host_version'] = opt1
+    return data
+
+def create_new_yaml(file, new_data):
+    with open(file, 'w') as stream:
+        yaml.dump(new_data, stream, default_flow_style=False, allow_unicode=True)
+
+def main():
+
+    parser = optparse.OptionParser()
+    parser.add_option("--branch-id",
+                      dest="release_branch",
+                      help="one of them : 1.7, 1.8 or 1.9")
+    parser.add_option("--acicontainers-tools",
+                      dest="acicontainers_tools",
+                      help="last successful build for AciContainers-tools job")
+    parser.add_option("--opflex-container",
+                      dest="opflex_container",
+                      help="last successful build for OpFlex-container")
+    parser.add_option("--ovs-container",
+                      dest="ovs_container",
+                      help="last successful build for OVS-container")
+
+    (opts, _) = parser.parse_args()
+    if (not opts.release_branch or not opts.acicontainers_tools or not opts.opflex_container or not opts.ovs_container):
+        parser.error("Required parameter(s) missing")
+
+    orig_data = read_yaml_file(VERSIONS_PATH)
```

Comment:
should we not have a null-check for VERSIONS_PATH ? Incase the relative path is wrong or versions.yaml is missing. This can be handled in a unit-test though, since I did not see the unit-test hence raising it

Reply:
Context manager does that for us. If `VERSIONS_PATH` file is not present then weget following type of error.
```
python
Python 2.7.10 (default, Oct  6 2017, 22:29:07)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> with open('foo', 'r') as file:
...     data = file.readlines()
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IOError: [Errno 2] No such file or directory: 'foo'
```